test/csuite/wt8290_rollback_reason/main.c

```diff
@@ -0,0 +1,79 @@
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+const char *rollback_error = "oldest pinned transaction ID rolled back for eviction";
+
+/*
+ * JIRA ticket reference: WT-4699 Test case description: Use a JSON dump cursor on a projection, and
+ * overwrite the projection string. Failure mode: On the first retrieval of a JSON key/value, a
+ * configure parse error is returned.
+ */
+int
+main(int argc, char *argv[])
+{
+    TEST_OPTS *opts, _opts;
+    WT_CURSOR *c;
+    WT_DECL_RET;
+    WT_SESSION *session;
+    const char *rollback_reason, *reason_2;
+    WT_UNUSED(c);
+    opts = &_opts;
+    memset(opts, 0, sizeof(*opts));
+    testutil_check(testutil_parse_opts(argc, argv, opts));
+    testutil_make_work_dir(opts->home);
+
+    testutil_check(wiredtiger_open(opts->home, NULL, "create,cache_size=1M", &opts->conn));
+    testutil_check(opts->conn->open_session(opts->conn, NULL, NULL, &session));
+    testutil_check(session->create(session, opts->uri, "key_format=i,value_format=S"));
+    testutil_check(session->open_cursor(session, opts->uri, NULL, NULL, &c));
+    rollback_reason = session->get_rollback_reason(session);
+    testutil_assert(rollback_reason == NULL);
+
+    testutil_check(session->begin_transaction(session, NULL));
+    for (int i = 0; i < 20000; i++) {
+        c->set_key(c, i);
+        c->set_value(c, "abcdefghijklmnopqrstuvwxyz");
+        ret = c->insert(c);
+        if (ret != 0) {
+            if (ret == WT_ROLLBACK) {
+                reason_2 = session->get_rollback_reason(session);
+                testutil_assert(strcmp(reason_2, rollback_error) == 0);
+                break;
+            } else
+                testutil_die(ret, "Unexpected error occurred while inserting values.");
+        }
+    }
+    testutil_check(session->rollback_transaction(session, NULL));
+    rollback_reason = session->get_rollback_reason(session);
+    testutil_assert(strcmp(rollback_reason, rollback_error) == 0);
+
+    testutil_check(session->close(session, NULL));
+    testutil_cleanup(opts);
+    return (EXIT_SUCCESS);
+}
```

Comment:
And, it's probably a lot faster and less fragile to use a write conflict instead of an eviction problem?
```
class test_xxx(wttest.WiredTigerTestCase):
    def test_xxx(self):
        uri = "table:xxx"
        ds = SimpleDataSet(self, "table:xxx", 10, key_format='S', value_format='S')
        ds.populate()

        s1 = self.session
        c1 = s1.open_cursor(uri)
        s2 = self.conn.open_session()
        c2 = s2.open_cursor(uri)
        s1.begin_transaction()
        c1[ds.key(5)] = "aaa"
        s2.begin_transaction()
        c2.set_key(ds.key(5))
        c2.set_value("bbb")
        msg = '/conflict between concurrent operations/'
        self.assertRaisesException(wiredtiger.WiredTigerError, lambda: c2.update(), msg)
        self.assertEquals('/' + s2.get_rollback_reason() + '/', msg)
```

Reply:
Agreed I'll write a python test instead.